src/pages/events.tsx

```diff
@@ -0,0 +1,404 @@
+        { name: "Teaching Adult Lessons", 
+            date: "2023-11-18",                
+            time: "5:00pm", 
+            description: "Assisting adults with tennis practice", 
+            slotsOpen: 7, 
+            totalSlots: 20, 
+            type: "participant"},
+        { name: "Teaching Adult Lessons", 
+            date: "2023-11-20",                
+            time: "5:00pm", 
+            description: "Assisting adults with tennis practice", 
+            slotsOpen: 7, 
+            totalSlots: 20, 
+            type: "participant"}, 
+        { name: "Court Cleanup", 
+            date: "2023-11-20", 
+            time: "8:00pm", 
+            description: "", 
+            slotsOpen: 7, 
+            totalSlots: 20, 
+            type: "volunteer"}, 
+        { name: "Teaching Adult Lessons", 
+            date: "2023-11-25",                
+            time: "3:00pm", 
+            description: "Assisting adults with tennis practice", 
+            slotsOpen: 7, 
+            totalSlots: 20, 
+            type: ["volunteer", "participant"]},    
+    ];
+
+    const upcomingEvents = events.filter((event) => {
+        return new Date(event.date).getDate() - new Date().getDate() >= 0;
+    });
+
+    const eventsToday = events.filter((event) => {
+        return new Date(event.date).getDate() - new Date().getDate() === -1;
+    });
+
+    const formatDateMonthDay = (eventDate: string) => {
+        const months = [
+            "January", "February", "March", "April", "May", "June",
+            "July", "August", "September", "October", "November", "December"
+        ];
+      
+        const dateObj = new Date(eventDate);
+        const month = months[dateObj.getUTCMonth()];
+        const day = dateObj.getUTCDate();
+      
+        return `${month} ${day}`;
+    };
+
+    const getClassName = (eventType: any) => {
+        if (Array.isArray(eventType)) {
+            return "upcoming-events-container-both";
+        } else {
+            return `upcoming-events-container-${eventType}`;
+        }
+    };
+
+    const listTypes = (eventType: any) => {
+        if (Array.isArray(eventType)) {
+            return "";
+        } else if (eventType === "volunteer") {
+            return "Volunteer";
+        } else {
+            return "Participant";
+        }
+    };
+
+    const handleRegisterClick = (eventNameTime) => {
+        setRegistrationStates(prev => ({ ...prev, [eventNameTime]: true }));
+    };
+    
+    const handleCancelClick = (eventNameTime) => {
+        setRegistrationStates(prev => ({ ...prev, [eventNameTime]: false }));
+    };
+
+    const getEventCountsByDay = (events) => {
+        const counts = {};
+        events.forEach((event) => {
+            const dateStr = new Date(event.date + "T00:00:00").toDateString();
+            counts[dateStr] = (counts[dateStr] || 0) + 1;
+        });
+        return counts;
+    };
+
+    const eventCounts = getEventCountsByDay(events);
```

Comment:
It would be more efficient to put this inside a `useEffect()` since you only need to run it once.